### PR TITLE
chore(devDeps): upgrade from metro 0.71 to 0.73

### PR DESCRIPTION
**Pull request by legobeat**
## **Description**

[Context](https://github.com/MetaMask/metamask-mobile/pull/7114#discussion_r1371227029).

Currently, three different versions of `metro` packages are being pulled in. Specifically, the explicitly depended on version is not even used.

This removes duplicates and aligns on the highest currently used version, `0.73.10`.

This implies a bump of `metro-config` to the matching version.

## **Related issues**

- #7595 
- #7114

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by cortisiko** [comment]
> Hmm, could this CI error be unrelated?

@legobeat
Triggering the E2E tests automatically is specifically supported on branches but not on forks. Can you open a branch inside the mobile repo instead of creating a fork branch?

**Comment by leotm** [comment]
great explanation again [earlier](https://github.com/MetaMask/metamask-mobile/pull/7114#discussion_r1375561810)

re-thinking let's go couple last steps further
- <s>move `metro-config` from _dep_ to _devDep_</s>
- (more thought) untrack it in package.json

so now fully/officially aligned on [RN core](https://github.com/facebook/react-native/blob/v0.71.14/template/package.json#L28) w simplest setup possible

then cleanup

https://github.com/MetaMask/metamask-mobile/blob/3022144b736727b8f15536d90a067087f85860ed/.depcheckrc.yml#L37

File: package.json [hunk 1]
```diff
@@ -253,7 +253,7 @@
     "lodash": "^4.17.21",
     "lottie-ios": "3.4.1",
     "lottie-react-native": "5.1.5",
-    "metro-config": "^0.71.1",
+    "metro-config": "~0.73.10",
```
Comment (leotm) [inline package.json:256]: > I do think `metro-config` could stay as a `devDependency` though - as it is imported explicitly in `metro.config.js`.
> 
> But thanks for reminding me on `.depcheckrc.yml` - updated.

ok yep fair point ^ let's move this from a dep to a devDep

then we're g2g after the conflicts caused by my Xcode 15 pr 😅

Comment (leotm) [reply package.json]: 👌

**Review by leotm — DISMISSED** [review]
🧹🚀

**Review by leotm — DISMISSED** [review]
ci / dupe in action 👌

worth adding to our husky step?

https://github.com/MetaMask/metamask-mobile/blob/867a410643e69cb3e81acb8f667436767e88daa5/package.json#L90-L94

edit: actually prob not, running this per commit would get annoying

**Review by leotm — DISMISSED** [review]

**Comment by Cal-L** [comment]
Do you mind providing a video of the app using the latest version of metro

**Comment by Cal-L** [comment]
Confirmed working - 
image

**Comment by Cal-L** [comment]
started E2E smoke tests - https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/bf829352-ad45-4897-8be8-0a906a441e88

**Comment by Cal-L** [comment]
Could you fix the invalid PR template check?

**Review by leotm — APPROVED** [review]

**Pull request merged**